spring-web/src/main/java/org/springframework/http/HttpHeaders.java

```diff
@@ -1626,4 +1631,12 @@ public static HttpHeaders readOnlyHttpHeaders(HttpHeaders headers) {
 		return (headers.readOnly ? headers : new HttpHeaders(headers, true));
 	}
 
+	/**
+	 * Return a {@code HttpHeaders} object that wraps an native headers map.
+	 */
+	public static HttpHeaders wrapNativeHeaders(MultiValueMap<String, String> nativeHeaders) {
```

Comment:
Not really sure this needs a static method. Personally, I think having a public `HttpHeaders(MultiValueMap<String, String> nativeHeaders`) constructor is good enough.